### PR TITLE
done 4

### PR DESCRIPTION
**Pull request by ThePotato97**

### CODE REVIEW

File: api/__tests__/app.test.js [hunk 1]
```diff
@@ -42,7 +43,46 @@ describe("/api", () => {
...
+          expect(article).toHaveProperty("body");
+          expect(article).toHaveProperty("topic");
+          expect(article).toHaveProperty("created_at");
+          expect(article).toHaveProperty("votes");
+          expect(article).toHaveProperty("article_img_url");
+        });
+    });
+    it("should return an error when the parameter is wrong", () => {
+      return request(app)
+        .get("/api/articles/fails")
```
Comment (pippaamy) [inline api/__tests__/app.test.js:67]: this should be a 400

File: api/controllers/articles.controller.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    return;
+  }
+  selectArticleById(article_id).then((article) => {
+    const firstArticle = article[0];
+    if (!firstArticle) {
+      next({ status: 404, msg: "Article ID does not exist" });
+      return;
+    }
+    res.status(200).send({ article: firstArticle });
+  });
```
Comment (pippaamy) [inline api/controllers/articles.controller.js]: can we add a catch block here instead

File: api/models/articles.model.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const db = require("../../db/connection.js");
+
+exports.selectArticleById = (id) => {
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = $1;`, [id])
+    .then(({ rows }) => rows)
+    .catch((err) => {
+      Promise.reject({ status: 500, msg: "Internal Server Error" });
```
Comment (pippaamy) [inline api/models/articles.model.js]: don't need to check for 500 it will have an automatic PSQL error handler that will do it for you

**Review by pippaamy — COMMENTED** [review]
great job, some changes to make!

**Comment by JVal-NC** [comment]
Pippas first comment for the test on line 65 should be a 400. Still does this using a custom error but psql will throw an error if the id is not a number. We should make use of these errors when possible.

**Review by JVal-NC — COMMENTED** [review]
Good work, go ahead and merge

**Pull request merged**